### PR TITLE
fix(amazonq): Improved LLM code review for file review

### PR DESCRIPTION
**Pull request by ashishrp-aws**
## Problem
- Code diff for file review format has changed in payload to improve LLM based code review accuracy and is dependent on git.

## Solution
- We are currently send entire file in diff format and run git commands to extract it. This change will remove git dependency for LLM based code review issues.

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be squash-merged at release time.
- Your code changes must meet the guidelines in [CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by justinmk3** [comment]
> Changelog is not needed as it is not customer facing bug/issue.

Then it's not a "fix". But based on your description, I can imagine a way to describe the improvement to customers, something like: "Improved performance of LLM based code review". 

Whenever there's any kind of improvement, it's worth mentioning. And conversely, if something is not being improved then it's not a "fix" or a "feat".

File: packages/core/src/codewhisperer/util/zipUtil.ts [hunk 1]
```diff
@@ -120,12 +120,8 @@ export class ZipUtil {
...
 
             if (scope === CodeWhispererConstants.CodeAnalysisScope.FILE_ON_DEMAND) {
-                await this.processCombinedGitDiff(
-                    zip,
-                    [workspaceFolder.uri.fsPath],
-                    uri.fsPath,
-                    CodeWhispererConstants.CodeAnalysisScope.FILE_ON_DEMAND
-                )
+                const gitDiffContent = `+++ b/${zipEntryPath}`
+                zip.addFile(ZipConstants.codeDiffFilePath, Buffer.from(gitDiffContent, 'utf-8'))
```
Comment (justinmk3) [inline packages/core/src/codewhisperer/util/zipUtil.ts:124]: is this covered by some sort of test?

Comment (ashishrp-aws) [reply packages/core/src/codewhisperer/util/zipUtil.ts:124]: Currently, code diff format is not covered by tests. I have to add tests for both project and file review. Will add them in followups.

File: packages/amazonq/.changes/next-release/Bug Fix-0c35b577-da17-4132-a0b7-e0d630e73cb9.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Bug Fix",
+	"description": "Imrpoved LLm code review for file review."
```
Comment (justinmk3) [inline packages/amazonq/.changes/next-release/Bug Fix-0c35b577-da17-4132-a0b7-e0d630e73cb9.json:3]:
```suggestion
	"description": "Improved LLM code review for file review."
```

File: packages/core/src/codewhisperer/util/zipUtil.ts (hunk 1 above)
Comment (ctlai95) [reply packages/core/src/codewhisperer/util/zipUtil.ts:124]: Can we add a comment explaining why we are doing this? It's hard to understand just by reading the code

Comment (ashishrp-aws) [reply packages/core/src/codewhisperer/util/zipUtil.ts:124]: Added comment and corrected this for windows as well.

**Review by ctlai95 — APPROVED** [review]

**Pull request merged**